### ISSUE TITLE
files_external:notify always stops after 60 seconds

### ISSUE THREAD

**Issue by goofacz** [icewind1991/files_inotify#11]
I experience problem with `files_inotify`. It correctly detects changes in local external storage, but it stops running after 60 seconds returning SUCCESS error code. I made a workaround and I automatically restart service.

Platform:
OS: Arch Linux
Nextcloud: 16.0.3.0
PHP: 7.3.8-1
nginx: 1.16.0-1
mariadb: 10.4.7-1

**Comment by pdschandler** [comment]
Yep. I've been reading these PR's and hoping by the time I had a chance to implement this I'd magically not be affected, but lo and behold - I'm seeing the same exact behavior. I'm on Alpine linux in the official Docker container with the added packages to support inotify. Adding the inotify scan commands into the supervisord config actually works pretty well to mitigate this issue - I made them always restart immediately, and while there is a small CPU spike every minute, it doesn't seem to affect performance after my testing. I also tested a lot of rapid file changes happening at once and I was never able to get the process to actually miss any of them, despite changing files as the process closed and restarted. So that's nice. But this isn't ideal.

Has anyone been able to figure out a likely cause for the process exiting? Seems very weird that it happens with a success code.

**Comment by Gatak** [comment]
Just got inotify installed and see the same issue with it exiting with success after a short while.

**Comment by tmolitor-stud-tu** [comment]
I just commented issue #7 on this bug.

**Comment by pdschandler** [comment]
This is fantastic, thank you so much. Hopefully we can see this merged and deployed to the app store.
